src/common/matrixproperties.jl

```diff
@@ -54,6 +61,86 @@ The `isapprox_kwargs` can be used to control the tolerances of the equality.
 See also [`isisometry`](@ref) and [`is_left_isometry`](@ref).
 """ is_right_isometry
 
-function is_right_isometry(A::AbstractMatrix; isapprox_kwargs...)
-    return isapprox(A * A', LinearAlgebra.I; isapprox_kwargs...)
+function is_right_isometry(A::AbstractMatrix; atol::Real = 0, rtol::Real = defaulttol(A), norm = LinearAlgebra.norm)
+    P = A * A'
+    nP = norm(P) # isapprox would use `rtol * max(norm(P), norm(I))`
+    diagview(P) .-= 1
+    return norm(P) <= max(atol, rtol * nP) # assume that the norm of I is `sqrt(n)`
+end
+
+"""
+    ishermitian(A; isapprox_kwargs...)
+
+Test whether a linear map is Hermitian, i.e. `A = A'`.
+The `isapprox_kwargs` can be used to control the tolerances of the equality.
+"""
+function ishermitian(A; atol::Real = 0, rtol::Real = 0, norm = LinearAlgebra.norm, kwargs...)
+    if iszero(atol) && iszero(rtol)
+        return ishermitian_exact(A; kwargs...)
+    else
+        return 2 * norm(project_antihermitian(A; kwargs...)) ≤ max(atol, rtol * norm(A))
+    end
+end
+function ishermitian_exact(A)
+    return A == A'
+end
+function ishermitian_exact(A::StridedMatrix; kwargs...)
+    return _ishermitian_exact(A, Val(false); kwargs...)
+end
+
+"""
+    isantihermitian(A; isapprox_kwargs...)
+
+Test whether a linear map is anti-Hermitian, i.e. `A = -A'`.
+The `isapprox_kwargs` can be used to control the tolerances of the equality.
+"""
+function isantihermitian(A; atol::Real = 0, rtol::Real = 0, norm = LinearAlgebra.norm, kwargs...)
+    if iszero(atol) && iszero(rtol)
+        return isantihermitian_exact(A; kwargs...)
+    else
+        return 2 * norm(project_hermitian(A; kwargs...)) ≤ max(atol, rtol * norm(A))
+    end
+end
+function isantihermitian_exact(A)
+    return A == -A'
+end
+function isantihermitian_exact(A::StridedMatrix; kwargs...)
```

Comment:
I think this is probably the right default, but we have to keep in mind that many strided arrays aren't subtyping `StridedMatrix`. (e.g. `Strided.StridedView <: StridedArray == false`), and several wrappers have a similar issue. 
Maybe we should rename the `_ishermitian_exact` functions to `strided_ishermitian_exact`, such that it is at least easy to patch that with package extensions?

Reply:
Can other types not simply overload `ishermitian_exact` and `isantihermitian_exact`? 

The `_ishermitian_exact` function is really an implementation detail (and is in fact not type restricted to strided matrices)

Reply:
What I mean is that the other types should overload `ishermitian_exact`, by making it call `strided_ishermitian_exact`, but it doesn't matter too much?